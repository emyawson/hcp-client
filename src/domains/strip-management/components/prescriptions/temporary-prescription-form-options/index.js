export * from './temporary-prescription-form-options.component';
