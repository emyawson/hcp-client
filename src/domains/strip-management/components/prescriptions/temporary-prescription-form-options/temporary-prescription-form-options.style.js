import styled from 'styled-components';

export const TemporaryPrescriptionRowDiv = styled.div`
  display: flex;
  flex-wrap: wrap;
`;
