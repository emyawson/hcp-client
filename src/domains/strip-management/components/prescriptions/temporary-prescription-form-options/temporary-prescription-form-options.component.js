import React from 'react';
import { always } from 'ramda';

import { LabelledDatePicker } from 'src/components/date-picker/labelled-date-picker';
import { LabelledDropdown } from 'src/components/dropdown/labelled-dropdown';
import { SelectContainerDiv } from 'src/domains/strip-management/scenes/prescription/prescription.style';
import { translate } from 'src/i18n';

import { TemporaryPrescriptionRowDiv } from './temporary-prescription-form-options.style';

export const TemporaryPrescriptionFormOptions = ({
  reasons,
  id,
  placeholderText,
  onDatesChange,
}) => (
  <TemporaryPrescriptionRowDiv>
    <SelectContainerDiv>
      <LabelledDatePicker
        label={translate('prescription.dateRange.label')}
        modelPath=".dateRange"
        onChange={(startDate, endDate) => {
          onDatesChange({ startDate, endDate });
        }}
        checkIfDateOutOfRange={() => always(false)}
        isPresetsPanelDisplayed={false}
      />
    </SelectContainerDiv>
    <SelectContainerDiv>
      <LabelledDropdown
        label={translate('prescription.reason.label')}
        modelPath=".reason"
        options={reasons[id]}
        placeholder={placeholderText}
      />
    </SelectContainerDiv>
  </TemporaryPrescriptionRowDiv>
);
