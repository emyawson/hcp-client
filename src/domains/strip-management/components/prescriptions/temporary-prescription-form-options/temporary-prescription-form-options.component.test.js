import React from 'react';
import { shallow } from 'enzyme';

import { TemporaryPrescriptionFormOptions } from './temporary-prescription-form-options.component';

describe('Temporary prescription from component', () => {
  it('Renders correctly', () => {
    const mockProps = {
      id: 'a1',
      reasons: { a1: [] },
      onDatesChange: () => {},
      placeholderText: 'Select',
    };
    const wrapper = shallow(
      <TemporaryPrescriptionFormOptions {...mockProps} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
