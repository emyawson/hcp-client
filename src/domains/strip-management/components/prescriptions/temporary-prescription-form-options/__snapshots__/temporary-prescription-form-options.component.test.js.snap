// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Temporary prescription from component Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    id="a1"
    onDatesChange={[Function]}
    placeholderText="Select"
    reasons={
      Object {
        "a1": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <Unknown
            checkIfDateOutOfRange={[Function]}
            isPresetsPanelDisplayed={false}
            label="Date Range"
            modelPath=".dateRange"
            onChange={[Function]}
          />
        </styled.div>,
        <styled.div>
          <LabelledDropdown
            disabled={false}
            label="Reason"
            modelPath=".reason"
            options={Array []}
            placeholder="Select"
          />
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Unknown
            checkIfDateOutOfRange={[Function]}
            isPresetsPanelDisplayed={false}
            label="Date Range"
            modelPath=".dateRange"
            onChange={[Function]}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "checkIfDateOutOfRange": [Function],
            "isPresetsPanelDisplayed": false,
            "label": "Date Range",
            "modelPath": ".dateRange",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <LabelledDropdown
            disabled={false}
            label="Reason"
            modelPath=".reason"
            options={Array []}
            placeholder="Select"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "label": "Reason",
            "modelPath": ".reason",
            "options": Array [],
            "placeholder": "Select",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <Unknown
              checkIfDateOutOfRange={[Function]}
              isPresetsPanelDisplayed={false}
              label="Date Range"
              modelPath=".dateRange"
              onChange={[Function]}
            />
          </styled.div>,
          <styled.div>
            <LabelledDropdown
              disabled={false}
              label="Reason"
              modelPath=".reason"
              options={Array []}
              placeholder="Select"
            />
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Unknown
              checkIfDateOutOfRange={[Function]}
              isPresetsPanelDisplayed={false}
              label="Date Range"
              modelPath=".dateRange"
              onChange={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "checkIfDateOutOfRange": [Function],
              "isPresetsPanelDisplayed": false,
              "label": "Date Range",
              "modelPath": ".dateRange",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <LabelledDropdown
              disabled={false}
              label="Reason"
              modelPath=".reason"
              options={Array []}
              placeholder="Select"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "label": "Reason",
              "modelPath": ".reason",
              "options": Array [],
              "placeholder": "Select",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
